src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -133,7 +199,33 @@ void IrcMessageHandler::addMessage(Communi::IrcMessage *_message,
         }
 
         auto msg = builder.build();
-        builder.triggerHighlights();
+
+        if (getSettings()->similarityEnabled)
+        {
+            bool isMyself =
+                msg->loginName ==
+                getApp()->accounts->twitch.getCurrent()->getUserName();
+            bool hideMyself = getSettings()->hideSimilarMyself;
+
+            if ((!isMyself || hideMyself) &&
```

Comment:
This type of if-case gets confusing quickly - if we move this scope (everything within `if similarityEnabled`) into a separate function we can control the logic with early outs instead, making the flow easier to understand

Reply:
moved to function